### PR TITLE
feat: Add pulsar module

### PR DESCRIPTION
**Pull request by smbecker**
## What does this PR do?

Add support for starting [Apache Pulsar](https://pulsar.apache.org/) containers.

## Why is it important?

This is required to write tests that target [Apache Pulsar](https://pulsar.apache.org/).

## How to test this PR

```sh
cd tests/Testcontainers.Pulsar.Tests
dotnet test
```

### CODE REVIEW

**Comment by entvex** [comment]
I believe this pull request is a duplicate. I’ve already submitted a pull request that adds Pulsar support. You can find it here: PR [#1103](https://github.com/testcontainers/testcontainers-dotnet/pull/1103).

**Review by eddumelendez — APPROVED** [review]

File: src/Testcontainers.Pulsar/PulsarBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        return new PulsarBuilder(new PulsarConfiguration(oldValue, newValue));
+    }
+
+    private async Task<bool> IsNodeReadyAsync(HttpResponseMessage response)
+    {
+        var content = await response.Content.ReadAsStringAsync()
+            .ConfigureAwait(false);
+
+        return "[\"standalone\"]".Equals(content, StringComparison.OrdinalIgnoreCase);
```
Comment (eddumelendez) [inline src/Testcontainers.Pulsar/PulsarBuilder.cs:78]: if `PULSAR_PREFIX_clusterName` env var is set then the wait strategy will fail. It should check the env var's value first.

**Review by eddumelendez — CHANGES_REQUESTED** [review]

File: src/Testcontainers.Pulsar/PulsarBuilder.cs (hunk 1 above)
Comment (HofmeisterAn) [reply src/Testcontainers.Pulsar/PulsarBuilder.cs:78]: As @entvex mentioned, there is already an open PR. We should focus on one implementation.

**Review by HofmeisterAn — COMMENTED** [review]
Thanks for the PR, but as mentioned, someone else already submitted an Apache Pulsar module. I will close this PR since we merged #1103.

**Pull request closed without merging**